src/Faker/Boolean.cs

```diff
@@ -1,20 +1,30 @@
-﻿using System;
-
-namespace Faker
+﻿namespace Faker
 {
-	/// <summary>
-	///   A collection of Boolean related resources.
-	/// </summary>
-	/// <threadsafety static="true" />
-	public static class Boolean
-	{
-		/// <summary>
-		///   Generates a random Boolean value
-		/// </summary>
-		/// <returns></returns>
-		public static bool Next()
-		{
-			return RandomNumber.Next(0, 2) == 0;
-		}
-	}
+    /// <summary>
+    ///   A collection of Boolean related resources.
+    /// </summary>
+    /// <threadsafety static="true" />
+    public static class Boolean
+    {
+        /// <summary>
+        ///   Generates a random Boolean value
+        /// </summary>
+        /// <returns>The random boolean value</returns>
+        public static bool Next()
+        {
+            return RandomNumber.Next(0, 2) == 0;
+        }
+
+        /// <summary>
+        ///   Generates a weighted Boolean value
+        /// </summary>
+        /// <param name="trueProbability">
+        ///   The probability to have a true value. Ranges from 0.0 to 1.0.
+        /// </param>
+        /// <returns>The random boolean value</returns>
+        public static bool Next(double trueProbability)
+        {
+            return RandomNumber.NextDouble() < trueProbability;
```

Comment:
Please add a check to see if the passed trueProbability is truly between 0.0 and 1.0, and throw an error if it is below or higher.

Reply:
Hi @AdmiringWorm. I'll do it, but... 
- guard conditions might throw exception even in case you pass exactly 0.0d or 1.0d (possibly as the result of a calculus)  because of floating point rounding errors;
- generation would simply always get false/true in case you pass <0 / >1 values, respectively, thus violating documented pre-conditions.

Let me know. 
Marcello.

Reply:
Thea reason I'm saying to add an guard condition, is just because you mention in the parameter documentation that it ranges from 0.0 to 1.0.

It's mainly to have both the documentation and the behaviour be the same, other than that it isn't important to me.

Reply:
Yes. Even with guard conditions, behavior might be incoherent with documentation, e.g. when you pass 1.0d and a rounding error let it turn to 1.00000000001d. And this error would be worse than a missing exception when you pass 100.0d as a parameter (violating the documented precondition) and get all `true` values. Ok, this last is a subtle bug, but IMO it is the lesser evil. 

I suggest to go whit this version. Obviously you are the maintainer and last word is yours. 

Cheers, 
Marcello

Reply:
Oh right, the rounding errors.
I didn't take into account for those.
Let me think about this one, I think we'll need to take a different approach in this case.
Maybe by using integers instead, and relying on modulus perhaps, I'll need to investigate that option first, or just doing a normal percentage based check without the modulus.

Reply:
I've changed my mind, it's fine as it is.
I'll merge it in as it is. If anything happens in the future, we can fix it then.

Reply:
Thanks @AdmiringWorm, and my best compliments for your DevOps skills: you set up a set of very nice features around your repository.

Reply:
@supix thanks alot, I try